--- before/cmake/FindBTOR2.cmake	2018-10-03 21:15:30.293613789 +0200
+++ after/cmake/FindBTOR2.cmake	2018-10-03 21:07:39.675590317 +0200
@@ -11,11 +11,7 @@
   set(BTOR2_ROOT_DIR ${PROJECT_SOURCE_DIR}/../btor2tools)
 endif()
 
-find_path(BTOR2_INCLUDE_DIR
-  NAMES btor2parser/btor2parser.h
-  PATHS "${BTOR2_ROOT_DIR}/src"
-  NO_DEFAULT_PATH
-  )
+find_path(BTOR2_INCLUDE_DIR NAMES btor2parser/btor2parser.h)
 
 # Prefer shared libraries if SHARED is enabled
 if(SHARED)
@@ -24,11 +20,7 @@
   set(libsuffix "a")
 endif()
 
-find_library(BTOR2_LIBRARIES
-  NAMES "libbtor2parser.${libsuffix}" btor2parser
-  PATHS "${BTOR2_ROOT_DIR}/build"
-  NO_DEFAULT_PATH
-  )
+find_library(BTOR2_LIBRARIES NAMES "libbtor2parser.${libsuffix}" btor2parser)
 
 include(FindPackageHandleStandardArgs)
 find_package_handle_standard_args(Btor2Tools BTOR2_INCLUDE_DIR BTOR2_LIBRARIES)
