--- before/cmake/FindBtor2Tools.cmake	2018-12-06 21:09:50.083346935 +0100
+++ after/cmake/FindBtor2Tools.cmake	2018-12-06 21:13:00.473356430 +0100
@@ -11,17 +11,9 @@
   set(Btor2Tools_ROOT_DIR ${PROJECT_SOURCE_DIR}/../btor2tools)
 endif()
 
-find_path(Btor2Tools_INCLUDE_DIR
-  NAMES btor2parser/btor2parser.h
-  PATHS "${Btor2Tools_ROOT_DIR}/src"
-  NO_DEFAULT_PATH
-  )
+find_path(Btor2Tools_INCLUDE_DIR NAMES btor2parser/btor2parser.h)
 
-find_library(Btor2Tools_LIBRARIES
-  NAMES btor2parser
-  PATHS "${Btor2Tools_ROOT_DIR}/build"
-  NO_DEFAULT_PATH
-  )
+find_library(Btor2Tools_LIBRARIES NAMES "libbtor2parser.${libsuffix}" btor2parser)
 
 include(FindPackageHandleStandardArgs)
 find_package_handle_standard_args(Btor2Tools
