--- before/cmake/FindLingeling.cmake	2018-12-06 21:18:02.292371484 +0100
+++ after/cmake/FindLingeling.cmake	2018-12-06 21:20:04.807377594 +0100
@@ -14,18 +14,9 @@
   set(GIVEN_Lingeling_ROOT_DIR FALSE)
 endif()
 
-find_path(Lingeling_INCLUDE_DIR
-  NAMES lglib.h
-  PATHS "${Lingeling_ROOT_DIR}"
-  NO_DEFAULT_PATH
-  )
-
-find_library(Lingeling_LIBRARIES
-  NAMES lgl
-  PATHS "${Lingeling_ROOT_DIR}"
-  NO_DEFAULT_PATH
-  )
+find_path(Lingeling_INCLUDE_DIR NAMES lglib.h)
 
+find_library(Lingeling_LIBRARIES NAMES "liblgl.${libsuffix}" lgl)
 
 include(FindPackageHandleStandardArgs)
 find_package_handle_standard_args(Lingeling
