--- before/makefile.in	2017-03-08 14:20:04.000000000 +0100
+++ after/makefile.in	2017-07-08 23:33:10.095627274 +0200
@@ -2,6 +2,8 @@
 
 CC=@CC@
 CFLAGS=@CFLAGS@
+OBJCFLAGS=@OBJCFLAGS@
+BINCFLAGS=@BINCFLAGS@
 
 LIBS=@LIBS@
 LDEPS=@LDEPS@
@@ -14,7 +16,7 @@
 -include makefile.mus
 -include makefile.other
 
-targets: liblgl.a
+targets: @LIBLGL@
 targets: lingeling plingeling ilingeling treengeling
 targets: lglmbt lgluntrace lglddtrace
 targets: @AIGERTARGETS@
@@ -22,26 +24,29 @@
 analyze:
 	clang --analyze $(CFLAGS) $(shell ls *.c *.h)
 
+liblgl.so: lglib.o lglbnr.o lgldimacs.o lglopts.o makefile $(EXTRAOBJS)
+	$(CC) -shared -Wl,-soname,liblgl.so -o $@ lglib.o lglbnr.o lgldimacs.o lglopts.o $(EXTRAOBJS)
+
 liblgl.a: lglib.o lglbnr.o lgldimacs.o lglopts.o makefile $(EXTRAOBJS)
 	ar rc $@ lglib.o lglbnr.o lgldimacs.o lglopts.o $(EXTRAOBJS)
 	ranlib $@
 
-lingeling: lglmain.o liblgl.a makefile $(LDEPS)
-	$(CC) $(CFLAGS) -o $@ lglmain.o -L. -llgl $(LIBS)
-plingeling: plingeling.o liblgl.a makefile $(LDEPS)
-	$(CC) $(CFLAGS) -pthread -o $@ plingeling.o -L. -llgl $(LIBS)
-treengeling: treengeling.o liblgl.a makefile $(LDEPS)
-	$(CC) $(CFLAGS) -pthread -o $@ treengeling.o -L. -llgl $(LIBS)
-blimc: blimc.o liblgl.a makefile $(AIGER)/aiger.o $(LDEPS)
-	$(CC) $(CFLAGS) -o $@ blimc.o -L. $(AIGER)/aiger.o -llgl $(LIBS)
-ilingeling: ilingeling.o liblgl.a makefile $(LDEPS)
-	$(CC) $(CFLAGS) -pthread -o $@ ilingeling.o -L. -llgl $(LIBS)
-lglmbt: lglmbt.o liblgl.a $(LDEPS)
-	$(CC) $(CFLAGS) -o $@ lglmbt.o -L. -llgl $(LIBS)
-lgluntrace: lgluntrace.o liblgl.a makefile $(LDEPS)
-	$(CC) $(CFLAGS) -o $@ lgluntrace.o -L. -llgl $(LIBS)
-lglddtrace: lglddtrace.o liblgl.a makefile $(LDEPS)
-	$(CC) $(CFLAGS) -o $@ lglddtrace.o -L. -llgl $(LIBS)
+lingeling: lglmain.o @LIBLGL@ makefile $(LDEPS)
+	$(CC) $(BINCFLAGS) -o $@ lglmain.o -L. -llgl $(LIBS)
+plingeling: plingeling.o @LIBLGL@ makefile $(LDEPS)
+	$(CC) $(BINCFLAGS) -pthread -o $@ plingeling.o -L. -llgl $(LIBS)
+treengeling: treengeling.o @LIBLGL@ makefile $(LDEPS)
+	$(CC) $(BINCFLAGS) -pthread -o $@ treengeling.o -L. -llgl $(LIBS)
+blimc: blimc.o @LIBLGL@ makefile $(AIGER)/aiger.o $(LDEPS)
+	$(CC) $(BINCFLAGS) -o $@ blimc.o -L. $(AIGER)/aiger.o -llgl $(LIBS)
+ilingeling: ilingeling.o @LIBLGL@ makefile $(LDEPS)
+	$(CC) $(BINCFLAGS) -pthread -o $@ ilingeling.o -L. -llgl $(LIBS)
+lglmbt: lglmbt.o @LIBLGL@ $(LDEPS)
+	$(CC) $(BINCFLAGS) -o $@ lglmbt.o -L. -llgl $(LIBS)
+lgluntrace: lgluntrace.o @LIBLGL@ makefile $(LDEPS)
+	$(CC) $(BINCFLAGS) -o $@ lgluntrace.o -L. -llgl $(LIBS)
+lglddtrace: lglddtrace.o @LIBLGL@ makefile $(LDEPS)
+	$(CC) $(BINCFLAGS) -o $@ lglddtrace.o -L. -llgl $(LIBS)
 
 lglmain.o: lglmain.c lglib.h makefile
 	$(CC) $(CFLAGS) -c lglmain.c
@@ -61,13 +66,13 @@
 	$(CC) $(CFLAGS) -c lglddtrace.c
 
 lglib.o: lglib.c lglib.h lglconst.h lgloptl.h lglopts.h makefile $(HDEPS)
-	$(CC) $(CFLAGS) -c lglib.c
+	$(CC) $(OBJCFLAGS) -c lglib.c
 lgldimacs.o: lgldimacs.c lgldimacs.h makefile
-	$(CC) $(CFLAGS) -c lgldimacs.c
+	$(CC) $(OBJCFLAGS) -c lgldimacs.c
 lglbnr.o: lglbnr.c lglcfg.h lglcflags.h makefile
-	$(CC) $(CFLAGS) -c lglbnr.c
+	$(CC) $(OBJCFLAGS) -c lglbnr.c
 lglopts.o: lglopts.c lglopts.h lglconst.h lgloptl.h makefile
-	$(CC) $(CFLAGS) -c lglopts.c
+	$(CC) $(OBJCFLAGS) -c lglopts.c
 
 lglcfg.h: VERSION mkconfig.sh lglbnr.c lglib.c lglmain.c lglcflags.h makefile
 	rm -f $@
@@ -84,6 +89,6 @@
 	rm -f lingeling plingeling ilingeling treengeling blimc
 	rm -f lglmbt lgluntrace lglddtrace
 	rm -f *.gcno *.gcda cscope.out gmon.out *.gcov *.gch *.plist
-	rm -f *.E *.o *.s *.a log/*.log 
+	rm -f *.E *.o *.s *.a *.so log/*.log 
 
 .PHONY: all targets clean clean-config clean-all
